### PR TITLE
Improve GroupingSet local plan

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Support colocate plan when the common column sub-list of grouping set satisfy property requirement

like sql: 
`select v1, v2 from t0 group by grouping sets ((v1, v2), (v1, v3), (v1));`

and t0 distribute by v1 column, can be get a colocate plan like:

```
  STREAM DATA SINK
    EXCHANGE ID: 03
    HASH_PARTITIONED: 1: v1, 2: v2, 3: v3, 4: GROUPING_ID

  2:AGGREGATE (update serialize)
  |  STREAMING
  |  group by: 1: v1, 2: v2, 3: v3, 4: GROUPING_ID
  |  
  1:REPEAT_NODE
  |  repeat: repeat 2 lines [[1, 2], [1, 3], [1]]
  |  
  0:OlapScanNode
     TABLE: t0
     PREAGGREGATION: ON
```

### CODE REVIEW

**Comment by HangyuanLiu** [comment]
Please explain the reason for your modification and what problem you are trying to solve

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 18 / 18 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/RequiredPropertyDeriver.java|14|14|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/OutputPropertyDeriver.java|3|3|100.00%|

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/GroupingSetTest.java [hunk 1]
```diff
@@ -121,6 +121,7 @@ public void testPushDownOverRepeatNode() throws Exception {
                 ") t\n" +
                 "WHERE IF(k2 IS NULL, 'ALL', k2) = 'ALL'";
         String plan = getFragmentPlan(sql1);
+        System.out.println(plan);
```
Comment (HangyuanLiu) [inline fe/fe-core/src/test/java/com/starrocks/sql/plan/GroupingSetTest.java:124]: delete

**Review by HangyuanLiu — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 18 / 18 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/RequiredPropertyDeriver.java|14|14|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/OutputPropertyDeriver.java|3|3|100.00%|

**Review by kangkaisen — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Pull request merged**